### ISSUE TITLE
Training ignores reward?

### ISSUE THREAD

**Issue by XBagon** [Unity-Technologies/ml-agents#416]
I'm trying to train a car, which can accelerate and steer, to reach a goal.
The input I give is from rays, they give distance to the walls on the left, right and front.
I'm really confused about the results of my trainings. Sometimes it manages to reach the goal and afterwards ignores it and even improves the negative reward to anti-perfection. It needed a lot of tuning of hyperparameter to even get the car to reach the goal once and now I'm just confused. There's a small negative reward for each step and a reward = 1000 for reaching the goal. Also the negative reward is smaller when the car has a higher velocity. 
Help is greatly appreciated!

![image](https://user-images.githubusercontent.com/1523292/37154468-6d73b3ce-22e0-11e8-9dc5-94c89e56e779.png)

**Comment by vincentpierre** [comment]
Hi, I think my best guess is that a reward of 1000 confuses the neural network which can only work with small values (between -1 and 1) I think you should have a look at [this page](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/best-practices.md#rewards). I think it is a problem with the environment, not the hyperparameters.
I hope this helps !

**Comment by XBagon** [comment]
Thanks for your answer! I changed the rewards to values in the range of [-1,1] and I'm not sure if it changed a lot. I also followed the tips from the page you linked. Now it doesn't look like it's just trying to forget everything it learned, but just not learning at all. Could be the short training process, but I imagine it should have better looking curves. 😎 I'm a bit desperate, I feel like I'm doing wrong something very essential. Also when I use the model in Unity, the car doesn't move at all, which is weird, cause it gets a higher reward by moving at all.
![image](https://user-images.githubusercontent.com/1523292/37288283-25e48680-2607-11e8-9168-5ebe3b1bb67e.png)

**Comment by awjuliani** [comment]
HI @XBagon,

The length of your episodes seems somewhat long. Can I ask what the `Frame Skip` you are using in the Academy is? You may want to increase it, if it is set to 0, as that forces the network to make decisions at a very high frequency, which can lead to learning issues.

**Comment by XBagon** [comment]
Hi @awjuliani. The episode length is that long, because I wanted to give the car enough time to finish the track even when doing some things wrong, so I set the max length to 3000, maybe this wasn't a good idea. The Frame Skip is set to 0, so I will try increasing that. Thanks for your help!

**Comment by mmattar** [comment]
Hi @XBagon - has your issue been resolved? Let us know if you need any additional support.

**Comment by XBagon** [comment]
Sorry for no further feedback. Honestly.. I gave up. It was a fun project and it was starting to get annoying and it felt like my changes had no impact at all. Will probably pick this up again, one day.

**Comment by xiaomaogy** [comment]
Hi @XBagon, we are closing this issue due to inactivity, feel free to open it if you need more help.

**Comment by icaro56** [comment]
@awjuliani  Where is this Frame Skip parameter?
